files/scripts/teamd.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+
+function debug()
+{
+    /usr/bin/logger $1
+    /bin/echo `date` "- $1" >> ${DEBUGLOG}
+}
+
+function check_warm_boot()
+{
+    SYSTEM_WARM_START=`$SONIC_DB_CLI STATE_DB hget "WARM_RESTART_ENABLE_TABLE|system" enable`
+    SERVICE_WARM_START=`$SONIC_DB_CLI STATE_DB hget "WARM_RESTART_ENABLE_TABLE|${SERVICE}" enable`
+    if [[ x"$SYSTEM_WARM_START" == x"true" ]] || [[ x"$SERVICE_WARM_START" == x"true" ]]; then
+        WARM_BOOT="true"
+    else
+        WARM_BOOT="false"
+    fi
+}
+
+function validate_restore_count()
+{
+    if [[ x"$WARM_BOOT" == x"true" ]]; then
+        RESTORE_COUNT=`$SONIC_DB_CLI STATE_DB hget "WARM_RESTART_TABLE|teamd" restore_count`
+        # We have to make sure db data has not been flushed.
+        if [[ -z "$RESTORE_COUNT" ]]; then
+            WARM_BOOT="false"
+        fi
+    fi
+}
+
+function check_fast_boot ()
+{
+    if [[ $($SONIC_DB_CLI STATE_DB GET "FAST_REBOOT|system") == "1" ]]; then
+        FAST_BOOT="true"
+    else
+        FAST_BOOT="false"
+    fi
+}
+
+start() {
+    debug "Starting ${SERVICE}$DEV service..."
+
+    check_warm_boot
```

Comment:
Should we also wait for database initialization before accessing it? I see that some of the service scripts use it, and some are missing it already.

https://github.com/Azure/sonic-buildimage/blob/master/files/scripts/swss.sh#L60
https://github.com/Azure/sonic-buildimage/blob/master/files/scripts/swss.sh#L137

Reply:
I saw this wait_for_database_service() is called from swss and syncd.  Teamd already has dependancy on swss.